src/main/java/teammates/common/datatransfer/TeamEvalResult.java

```diff
@@ -215,9 +219,11 @@ private static boolean isSpecialValue(int value) {
         double[] returnValue = new double[filterArray.length];
         for (int i = 0; i < filterArray.length; i++) {
             int filterValue = (int) filterArray[i];
-            boolean isSpecialValue = !isSanitized(filterValue)
-                    || filterValue == NA;
-            returnValue[i] = isSpecialValue ? NA : valueArray[i];
+            if (filterValue == NA || filterValue == NSU || !isSanitized(filterValue)) {
+                returnValue[i] = filterValue == NSU ? NSU : NA;
```

Comment:
This rewritten conditional does not seem to value add:
- `filterValue == NA` => `returnValue[i] = NA`
- `filterValue == NSU` => `returnValue[i] = NSU`
- `!isSanitized(filterValue)` => `returnValue[i] = NA`

That is already what the previous conditional does.

Reply:
@wkurniawan07 I think this happened previously <ul> `filterValue == NSU` => `returnValue[i] = valueArray[i]`</ul>

Reply:
Noted on that, apologies as it looks like I read the array names wrongly.